ros2/kachaka_grpc_ros2_bridge/src/component/diagnostics_component.cpp

```diff
@@ -0,0 +1,147 @@
+//  Copyright 2023 Preferred Robotics, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//     http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+
+#include <diagnostic_msgs/msg/diagnostic_array.hpp>
+#include <diagnostic_msgs/msg/diagnostic_status.hpp>
+#include <diagnostic_msgs/msg/key_value.hpp>
+#include <grpcpp/grpcpp.h>
+#include <nlohmann/json.hpp>
+#include <rclcpp/rclcpp.hpp>
+#include <rclcpp_components/register_node_macro.hpp>
+
+#include "../converter/ros_header.hpp"
+#include "../grpc_bridge.hpp"
+#include "../ros2_topic_bridge.hpp"
+#include "../stub_util.hpp"
+#include "kachaka-api.grpc.pb.h"
+
+namespace kachaka::grpc_ros2_bridge {
+using json = nlohmann::json;
+
+class DiagnosticsComponent : public rclcpp::Node {
+ public:
+  explicit DiagnosticsComponent(const rclcpp::NodeOptions& options)
+      : Node("diagnostics", options) {
+    stub_ = GetSharedStub(declare_parameter("server_uri", ""));
+
+    rclcpp::SensorDataQoS qos;
```

Comment:
Diagnosticsは到達性が大事なのでSystemDefaultで良いかと思います。

```suggestion
    rclcpp::SystemDefaultsQoS qos;
```

Reply:
@terakoji-pfr ご確認ありがとうございました。確かにそうですね。修正してpushしました。